src/module/functions/mod.rs

```diff
@@ -212,6 +219,11 @@ impl Module {
         }
         let num_imports = self.funcs.arena.len() - (amt as usize);
 
+        // First up serially create corresponding `LocalId` instances for all
+        // functions as well as extract the operators parser for each function.
+        // This is pretty tough to parallelize, but we can look into it later if
+        // necessary and it's a bottleneck!
```

Comment:
We would need to either make each function have a separate arena of locals, or put the local arena in a mutex (which would kind of defeat the purpose)

Reply:
Yeah I think if this becomes a bottleneck we have a few ways to handle it. We need to persist the mapping of index to id beyond function parsing to handle the local names section, but I think we could have "bulk insert" operations where the arena is built locally per-function and then inserted into a map at the end.

I think we can actually go a pretty long way avoiding any synchronization beyond rayon, and ideally adding rayon doesn't add too much complexity in this crate